src/visitPage.js

```diff
@@ -89,37 +91,12 @@ class VisitPage extends Component {
       // { completed: false, active: false, title: 'Housing', form: HousingStep },
       { completed: false, active: false, title: 'Results', form: ResultsGraph }
     ];  // end this.steps {}
-
-    this.stepProps = {
-      currentStep:  this.state.currentStep,
-      nextStep:     this.nextStep,
-      previousStep: this.previousStep,
-      storeComplex: this.storeComplex, // Maybe put these straight on state
-      storeChecked: this.storeChecked, // Maybe put these straight on state
-      saveForm:     this.saveForm,
-      pageState:    this.state
-    };
-
   };  // End constructor()
 
-  storeChecked = (e, { name, checked }, callback) => {
-    var truth = this;
-    truth.setState({ [name]: checked },
-      function () {
-       // console.log( name, checked, truth );
-       if ( callback ) { callback( truth ); }
-    });
-  }
-
-  storeComplex = (e, { name, value }, callback) => {
-    var truth = this;
-    truth.setState(
-      { [name]: value },
-      function () {
-        // console.log( truth );
-        if ( callback ) { callback( truth ); }
-      }  // This is given no arguments
-    );
+  setClientProperty = (e, data) => {
+    let propertyName = data.name
+    let value = (data.checked != null && data.checked != undefined) ? data.checked : data.value  //This handles both complex values and checked values
```

Comment:
If you're going to have conditionals, see if `!==` will work. I'm not sure why `checked` would ever be 0, though. That would be very strange, and wrong, but strange and wrong things do happen.

Reply:
checked could be `false`, which means the value `false` exists and should be stored. I'm changing `!=` to `!==`, though.

Reply:
Wow, brain must have been out to lunch. That's a great point. In that case though, would you not simply check that the type of `checked` is a boolean?